pkg/logs/input/docker/tailer.go

```diff
@@ -170,6 +170,13 @@ func (t *Tailer) readForever() {
 			n, err := t.read(inBuf, readTimeout)
 			if err != nil { // an error occurred, stop from reading new logs
 				switch {
+				case isReaderClosed(err):
+					// The reader has been closed during the shut down process of the tailer
+					// because as stated in the code base of docker:
+					// "It's up to the caller to close the stream.".
```

Comment:
It is not easy to grasp that when the reader stops, it return a `http: read on closed response body` error, which signals to the tailer that it should stop.